include/swift/Remote/MetadataReader.h

```diff
@@ -29,6 +30,28 @@
 namespace swift {
 namespace remote {
 
+template <typename BuiltType> class FunctionParam {
+  BuiltType Type;
+  StringRef Label;
+  ParameterTypeFlags Flags;
```

Comment:
@DougGregor please let me know what you think about this? I'm not sure if it's comme il faut to use structures from `AST/Types.h` in here.